heareval/tasks/pipeline.py

```diff
@@ -284,18 +317,49 @@ def get_metadata(self):
         )
         return metadata
 
+    def subsample_metadata(self, metadata, max_files):
+
+        # Get group count for each group
+        grp_count = metadata["stratify_key"].value_counts() * max_files / len(metadata)
+        # Sort by the split key and the subsample key
+        metadata = metadata.sort_values(
+            by=["split_key", "subsample_key"], ascending=[True, True]
+        )
+        # Groupby and select the required sample from each group
+        sampled_metadata = pd.concat(
+            [
+                # Ensure at least 1 sample is selected for each group
+                stratify_grp.head(1 if grp_count[grp] < 1 else int(grp_count[grp]))
+                for grp, stratify_grp in metadata.groupby("stratify_key")
+            ]
+        )
+        # Do some assertions
+        assert set(sampled_metadata["stratify_key"].unique()) == set(
+            metadata["stratify_key"].unique()
+        ), "All stratify groups are not in the sampled metadata."
+        # Add the num of groups here for handling addition in case of highly
+        # imbalanced subsampling
+        assert len(sampled_metadata) <= max_files + len(
+            grp_count
+        ), "Sampled metadata is more than the allowed max files + unique groups"
```

Comment:
Checks
- if all the labels are there in the method
- if the subsampled data points are more than the `max_subsample + len(grp_count)`. The length of group count is here since some groups might be too small and we might need to take one sample for the group.

Reply:
Same, this should be a comment